tests/aws/services/stepfunctions/v2/mocking/test_base_scenarios.py

```diff
@@ -1,28 +1,174 @@
+            template["States"]["step1"]["Resource"] = (
+                f"arn:aws:lambda:us-east-1:111111111111:function:{function_name}"
+            )
+            definition = json.dumps(template)
+            create_and_record_mocked_execution(
+                aws_client,
+                create_state_machine_iam_role,
+                create_state_machine,
+                sfn_snapshot,
+                definition,
+                exec_input,
+                state_machine_name,
+                test_name,
+            )
+
+    @markers.aws.only_localstack
+    def test_lambda_invoke_retries(
+        self,
+        aws_client,
+        create_state_machine_iam_role,
+        create_state_machine,
+        monkeypatch,
+        mock_config_file,
+    ):
+        template = ScenariosTemplate.load_sfn_template(
+            ScenariosTemplate.LAMBDA_INVOKE_WITH_RETRY_BASE
+        )
+        template["States"]["InvokeLambdaWithRetry"]["Resource"] = (
+            "arn:aws:lambda:us-east-1:111111111111:function:nosuchfunction"
+        )
+        definition = json.dumps(template)
+
+        state_machine_name = f"mocked_state_machine_{short_uid()}"
+        test_name = "TestCaseName"
+        lambda_not_ready_timeout_200_string_body = MockedResponseLoader.load(
+            MockedResponseLoader.LAMBDA_NOT_READY_TIMEOUT_200_STRING_BODY
+        )
+        mock_config = {
+            "StateMachines": {
+                state_machine_name: {
+                    "TestCases": {
+                        test_name: {
+                            "InvokeLambdaWithRetry": "lambda_not_ready_timeout_200_string_body"
+                        }
+                    }
+                }
+            },
+            "MockedResponses": {
+                "lambda_not_ready_timeout_200_string_body": lambda_not_ready_timeout_200_string_body
+            },
+        }
+        mock_config_file_path = mock_config_file(mock_config)
+        monkeypatch.setattr(config, "SFN_MOCK_CONFIG", mock_config_file_path)
+
+        role_arn = create_state_machine_iam_role(target_aws_client=aws_client)
+
+        state_machine = create_state_machine(
+            target_aws_client=aws_client,
+            name=state_machine_name,
+            definition=definition,
+            roleArn=role_arn,
+        )
+        state_machine_arn = state_machine["stateMachineArn"]
+
+        sfn_client = aws_client.stepfunctions
+        execution = sfn_client.start_execution(
+            stateMachineArn=f"{state_machine_arn}#{test_name}", input="{}"
+        )
+        execution_arn = execution["executionArn"]
+
+        await_execution_terminated(stepfunctions_client=sfn_client, execution_arn=execution_arn)
+
+        execution_history = sfn_client.get_execution_history(
+            executionArn=execution_arn, includeExecutionData=True
+        )
+        events = execution_history["events"]
+
+        event_4 = events[4]
+        assert event_4["taskFailedEventDetails"] == {
```

Comment:
praise: neat advanced scenario 💯 

sidenote: This shows how cumbersome parity testing (against StepFunction Local) is without snapshots 😅